raft/server.cc

```diff
@@ -469,15 +469,16 @@ future<> server_impl::wait_for_entry(entry_id eid, wait_type type, seastar::abor
 
 future<entry_id> server_impl::add_entry_on_leader(command cmd, seastar::abort_source* as) {
     // Wait for sufficient memory to become available
+    semaphore_units<> memory_permit;
     try {
-        co_await _fsm->consume_memory(as, log::memory_usage_of(cmd));
+        memory_permit = co_await _fsm->wait_for_memory_permit(as, log::memory_usage_of(cmd));
     } catch (semaphore_aborted&) {
```

Comment:
Not related to your patch, but why do we eat all exceptions but `semaphore_aborted`. Should we rethrow them as is?

Reply:
didn't get it, we eat only `semaphore_aborted`, because it's an implementation detail, `request_aborted` is a part of api

Reply:
Yes, my bad. We path trough other exceptions.